frontend/src/components/worksheets/Worksheet/Worksheet.js

```diff
@@ -1708,13 +1708,13 @@ class Worksheet extends React.Component {
         let newSource = [...this.state.ws.info.source];
 
         validBundles.forEach((index, i) => {
-            const items = ['% display contents ' + path, newSource[index], ''];
-            // For each checked bundle, we are adding 3 new lines to the source right after each checked bundle
-            newSource = [
-                ...newSource.slice(0, index + 3 * i + 1),
-                ...items,
-                ...newSource.slice(index + 3 * i + 1),
-            ];
+            newSource.splice(
```

Comment:
@yuqijin Oh, I meant `newSource = [...newSource].splice(...`. This way, we're still immutably changing `newSource`.

Reply:
I don't think we need to do 'newSource=...' since the splice function will change the original array (while slice function does not)

Reply:
Oh, you're right -- sorry, didn't realize that `newSource` was already immutably created from `this.state.ws.info.source`.